auth/impersonate/idtoken.go

```diff
@@ -56,6 +57,19 @@ type IDTokenOptions struct {
 	Client *http.Client
 }
 
+func (o *IDTokenOptions) validate() error {
+	if o == nil {
+		return errors.New("impersonate: options must be provided")
```

Comment:
Should this be labeled "idtoken"? Or even if "impersonate" is the surface, should we somehow identify the `IDTokenOptions` origin of this error?

Reply:
you only see this from calling NewIDTokenProvider, so it should be implied. The idom is prefix with the package so that is why it is impersonate.